### PR TITLE
feat: object conversion for readonly charts (#1335)

### PR DESCRIPTION
**Pull request by Caele**
## Motivation

I decided to make this simpler for now just add a parameter to the current api to support patching and keep the new API hidden.

## Requirements checklist

- [x] Api specification
  - [x] Ran `yarn spec`
    - [x] No changes **_OR_** API changes has been formally approved
- [x] Unit/Component test coverage
- [x] Correct PR title for the changes (fix, chore, feat)

When build and tests have passed:

- [ ] Add code reviewers, for example @qlik-oss/nebula-core

### CODE REVIEW

File: apis/nucleus/src/viz.js [hunk 1]
```diff
@@ -106,27 +109,43 @@ export default function viz({ model, halo, initialError, onDestroy = async () =>
       unmountCell = noopi;
     },
     /**
-     * Converts the visualization to a different registered type
+     * Converts the visualization to a different registered type. Will update properties if permissions allow, else will patch.
+     *
+     * Not all chart types are compatible, similar structures are required.
+     *
+     * NOTE: Consider using viz.convert.toType instead for session based conversion
```
Comment (veinfors) [inline apis/nucleus/src/viz.js:116]: Maybe we should wait with referencing `viz.convert` until it's part of the api?
Also would be nice to clarify why the user should consider using that instead... some specific benefit?

File: apis/supernova/src/index.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import generator from './generator';
+import JSONPatch from './json-patch';
 
-export { generator };
+export { generator, JSONPatch };
```
Comment (veinfors) [inline apis/supernova/src/index.js:4]: Feels a little bit weird, the way we import and then export generator and and JSONPatch like this. Maybe we should consider creating some kind of common internal pkg instead for shared stuff, but we don't have to do it right now.

**Review by veinfors — APPROVED** [review]

File: apis/nucleus/src/viz.js (hunk 1 above)
Comment (Caele) [reply apis/nucleus/src/viz.js:116]: yes yes, i'll remove that part

File: apis/supernova/src/index.js (hunk 2 above)
Comment (Caele) [reply apis/supernova/src/index.js:4]: it is a bit weird, but supernova is an internal package, so no too far off.

**Pull request merged**